### DISCUSSION TITLE
Combination required aes

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#56] [Q&A]
Just wondered if there is a way to express different requirement combo options.  For the maps stuff this is relevant where a code like fips can uniquely id a US county as well as state name + county name combo.  So required aes "fips|(countyname&state)".  There are counties w same names across states. I'm guessing this isn't allowed but wanted to check, express interest.

**Comment by teunbrand** [comment]
If you have more complicated requirements, you can earmark all aesthetics participating in the rule as `optional_aes` or give it `NULL` default aesthetics. Then in `setup_params()` you figure out if the requirements are met and throw an error if they aren't.

**Reply by EvaMaeRey** [reply]
Ah nice! Thanks for the information. Does a Stat using this approach come to mind for reference? 🙏

**Reply by teunbrand** [reply]
StatEcdf$setup_params does this to integrate x/y fed through aesthetics and params

**Reply by EvaMaeRey** [reply]
https://github.com/tidyverse/ggplot2/blob/HEAD/R/stat-ecdf.R  I didn't see optional_aes used, but do see that conditional logic and warning in setup params.  I guess in the my case, I should ideally (eventually) go with optional_aes = c("fips", "county_name", "state_name") and then lay out unique logic and erroring in setup params...  Thank you!

**Reply by teunbrand** [reply]
That does sound like the right path to me :)
